### PR TITLE
Support performance mode of GLM4 model

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Support performance mode of GLM4 model

-  [x] Fit with `transformers 4.45`
-  [x] FIx output issue

PR validation: https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/11889194939

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/speculative.py [hunk 1]
```diff
@@ -510,7 +510,7 @@ def _crop_past_key_values(self, past_key_values, new_cache_size, _enable_ipex=Fa
                 for k, v in past_key_values
             ]
         elif self.config.model_type == "chatglm":
-            if self.config.num_layers == 40 and hasattr(self.config, 'rope_ratio'):
+            if self.config.num_layers in [28, 40] and hasattr(self.config, 'rope_ratio'):
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/speculative.py:513]: add comments (on which models these conditions are intended for)

File: python/llm/src/ipex_llm/transformers/models/chatglm4.py [hunk 2]
```diff
@@ -76,9 +76,14 @@ def chatglm4_model_forward(
     if full_attention_mask is None:
         if (attention_mask is not None and not attention_mask.all()) or\
                 (past_key_values and seq_length != 1):
-            full_attention_mask = self.get_masks(inputs_embeds,
-                                                 past_key_values,
-                                                 padding_mask=attention_mask)
+            if self.config.hidden_size == 4096:
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/models/chatglm4.py:79]: add comments (on which models these conditions are intended for)

File: python/llm/src/ipex_llm/transformers/speculative.py [hunk 3]
```diff
@@ -768,7 +768,10 @@ def _non_cpu_ipex_verify(self, verify_input_ids, past_key_values, cur_attention_
         forward_args["attention_mask"] = cur_attention_mask
 
     if self.config.model_type == "chatglm":
-        past_key_value_len = past_key_values[0][0].shape[0]
+        if self.config.num_layers in [28, 40] and hasattr(self.config, 'rope_ratio'):
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/speculative.py:771]: add comments (on which models these conditions are intended for)

File: python/llm/src/ipex_llm/transformers/speculative.py (hunk 1 above)
Comment (Oscilloscope98) [reply python/llm/src/ipex_llm/transformers/speculative.py]: Thank you for the comment. Have updated :)

**Review by cyita — APPROVED** [review]
LGTM

**Pull request merged**